core/src/services/gcs/backend.rs

```diff
@@ -66,6 +66,10 @@ pub struct GcsConfig {
     pub predefined_acl: Option<String>,
     /// The default storage class used by gcs.
     pub default_storage_class: Option<String>,
+    /// Explicitly disable authentication
+    ///
+    /// Used for testing purposes against storage emulators.
+    pub no_authentication: bool,
```

Comment:
Compared to adding explicit support for no authentication, I prefer having `allow_anonymous` combined with `disable_config` and `disable_vm_metadata`.

This provides users more control on the behavior they want and align with the existing s3 service config.